apis/istio/v1alpha3/serviceentry_types.go

```diff
@@ -0,0 +1,390 @@
+package v1alpha3
+/*
+Portions Copyright 2017 The Kubernetes Authors.
+Portions Copyright 2018 Adobe.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VirtualService is a Istio VirtualService resource
+type ServiceEntry struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ServiceEntrySpec `json:"spec"`
+}
+
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VirtualServiceList is a list of VirtualService resources
+type ServiceEntryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []ServiceEntry `json:"items"`
+}
+
+// Location specifies whether the service is part of Istio mesh or
+// outside the mesh.  Location determines the behavior of several
+// features, such as service-to-service mTLS authentication, policy
+// enforcement, etc. When communicating with services outside the mesh,
+// Istio's mTLS authentication is disabled, and policy enforcement is
+// performed on the client-side as opposed to server-side.
+type ServiceEntry_Location int32
+
+const (
+	// Signifies that the service is external to the mesh. Typically used
+	// to indicate external services consumed through APIs.
+	ServiceEntry_MESH_EXTERNAL ServiceEntry_Location = 0
```

Comment:
Doing so certainly makes this entire directory unmaintainable i.e.  Changing protobuf naming.

All the other files in this folder even contain the protobuf comments.  Why would you change that??

In this particular case, I think the `const` can be dropped completely.
For completeness, see [serviceentry.pb.go#L28](https://github.com/istio/api/blob/1.0.5/networking/v1alpha3/service_entry.pb.go#L28)

Reply:
In that case, please add some comments to code blocks that were copied from code generated by proto-compiler, since there is more than one ways to generate these `*_types.go` file.